src/services/adapters/ironbank.ts

```diff
@@ -86,10 +68,11 @@ export class IronBankAdapter<T extends ChainId> extends ContractService<T> {
    * @returns static
    */
   async assetsStatic(addresses?: Address[], overrides: CallOverrides = {}): Promise<IronBankMarketStatic[]> {
+    const contract = await this.contract;
     if (addresses) {
-      return await this.contract.read["assetsStatic(address[])"](addresses, overrides).then(structArray);
+      return await contract.read["assetsStatic(address[])"](addresses, overrides).then(structArray);
```

Comment:
We don't really need to await a return, since it's going to return the promise anyway, I think this applies to most of the call we have

Reply:
Fixed